float32 vMax
float32 rMax
